nipype/algorithms/confounds.py

```diff
@@ -300,16 +324,44 @@ class FramewiseDisplacement(BaseInterface):
         'tags': ['method']
     }]
 
-    def _run_interface(self, runtime):
+    def _run_interface(self, runtime, metric='L1'):
         mpars = np.loadtxt(self.inputs.in_file)  # mpars is N_t x 6
         mpars = np.apply_along_axis(
             func1d=normalize_mc_params,
             axis=1,
             arr=mpars,
             source=self.inputs.parameter_source)
-        diff = mpars[:-1, :6] - mpars[1:, :6]
-        diff[:, 3:6] *= self.inputs.radius
-        fd_res = np.abs(diff).sum(axis=1)
+
+        if self.inputs.metric == 'L1':
+            diff = mpars[:-1, :6] - mpars[1:, :6]
+            diff[:, 3:6] *= self.inputs.radius
+            fd_res = np.abs(diff).sum(axis=1)
+
+        elif self.inputs.metric == 'riemannian':
+            # TODO(nina): Check SPM convention for Euler angles
+            # (Tait-Bryan angles)
+            so3pars = mpars[:, DIM_TRANSLATIONS:]
+            so3pars = SO3_GROUP.rotation_vector_from_tait_bryan_angles(
+                so3pars,
+                extrinsic_or_intrinsic='intrinsic',
+                order='xyz')
+
+            se3pars = np.hstack(
+                [so3pars, mpars[:, :DIM_TRANSLATIONS]])
+
+            diag_rotations = self.inputs.radius * np.ones(DIM_ROTATIONS)
+            diag_translations = np.ones(DIM_TRANSLATIONS)
+            diag = np.concatenate([diag_rotations, diag_translations])
+            inner_product = np.diag(diag)
+            metric = InvariantMetric(
+                       group=SE3_GROUP,
+                       inner_product_mat_at_identity=inner_product,
+                       left_or_right='left')
+            fd_res = metric.dist(se3pars[:-1], se3pars[1:])
+
+        else:
+            raise ValueError(
+                'The parameter \'metric\' should be \'L1\' or \'riemannian\'.')
```

Comment:
By using the `Enum` trait with `mandatory=True` above, the only possible values are `'L1'` or `'riemannian'`, so we can drop this final `else`.